### ISSUE TITLE
Danish Holidays

### ISSUE THREAD

**Issue by jkoinnovatedk** [sql-bi/Bravo#793]
### Describe the problem

Is it by purpose that you have removed the earlier implemented Danish Holiday names in the template files?

### Bravo version

1.0.8 (1.0.8868.14547)

### Tabular model

Power BI Desktop, import mode

### Power BI

Power BI Desktop June 2024

**Comment by albertospelta** [comment]
Hi @jkoinnovatedk, could you please let me know which Danish holiday names were removed or point me to the commit where this change was introduced? Perhaps you are referring to this commit: https://github.com/sql-bi/Bravo/commit/8def2dbac584cbf96c82012dbcc0ff88ff08f1b5 where `Ferienavn` has been changed to `Helligdag`?

**Comment by jkoinnovatedk** [comment]
![image](https://github.com/sql-bi/Bravo/assets/33174586/0bf53c77-0e2e-46ad-8a07-4b19fdb392a8)

**Comment by albertospelta** [comment]
I have reviewed the commit history, and I did not find any changes to the template files that altered the translation. It appears that the names have consistently been in English since the introduction of Danish holidays. Nevertheless, we can proceed with translating them and releasing a new version to address the issue. I have started the translation [here](https://github.com/sql-bi/Bravo/pull/794/files) using an automatic translator. Could you please review them and confirm if the translations are accurate? Thanks

**Comment by albertospelta** [comment]
@jkoinnovatedk By the end of next week, we would like to release a new version that includes the updated translations of the Danish holiday names. If possible, it would be very helpful if you could review and confirm the accuracy of the translations we generated using an automatic translator here https://github.com/sql-bi/Bravo/pull/794/files . Thanks!

**Comment by jkoinnovatedk** [comment]
Hi Alberto

Sorry, got away from this subject

Official holidays:


  *   Nytårsdag
  *   Palmesøndag (the Sunday before Skærtorsdag (Mandy Thursday)
  *   Skærtorsdag
  *   Langfredag
  *   Påskedag (the Sunday before Anden påskedag (Easter Monday)
  *   Anden påskedag (some people write is like “2. påskedag”)
  *   Store bededag (only reel holiday until 2023 (included) since it has been “cancelled” by the government in Denmark from 2024 and forward but kept in Greenland and Faroe Islands – but may return with a new government)
  *   Kristi himmelfartsdag
  *   Pinsedag (the Sunday before Anden pinseday)
  *   Anden pinsedag
  *   Juledag (”Første juledag” or ”1. juledag”)
  *   Anden juledag (“2. Juledag”)

Many banks and some companies are also closed (but not official holidays):

  *   Grundlovsdag (5th of June)
  *   Juleaftensdag (Juleaften is the evening, and can be used (aften=evening))
  *   Nytårsaftensdag (again, Nytårsaften is the evening, and can be used

Ps. Most people don’t go to work Juleaftensdag and nytårsaftensdag since the unions has negotiated these two days as paid days off but it is not a holiday


Labour Day:

  *   1. maj (Labour Day or “Arbejdernes kampdag” is not a holiday in Denmark – you normally still have to go to work but some unions have negotiated this as a day off “if you go and demonstrate”)

Med venlig hilsen
Innovate Technology

Jørgen Koch
Microsoft Certified Trainer


[innovatelogo-mail]<http://www.innovate.dk/>

Mail      ***@***.******@***.***>
Telefon +45 70 23 00 14
Mobil    +45 20 23 00 14

Se mere på http://www.innovate.dk/

***@***.***

Fra: Alberto Spelta ***@***.***>
Sendt: 21. juli 2024 19:37
Til: sql-bi/Bravo ***@***.***>
Cc: Jørgen Koch ***@***.***>; Mention ***@***.***>
Emne: Re: [sql-bi/Bravo] Danish Holidays (Issue #793)


@jkoinnovatedk<https://github.com/jkoinnovatedk> By the end of next week, we would like to release a new version that includes the updated translations of the Danish holiday names. If possible, it would be very helpful if you could review and confirm the accuracy of the translations we generated using an automatic translator here https://github.com/sql-bi/Bravo/pull/794/files . Thanks!

**Comment by albertospelta** [comment]
Thank you for the information you provided. We have updated the names of the currently defined holidays and set 2023 as the last year for Store Bededag. At the moment, there are three holidays that are not included, and we have opened a dedicated issue [here](https://github.com/sql-bi/DaxTemplate/issues/53) for these as they likely require a change to the calculation.

For completeness, below is the resulting holiday configuration. If you encounter any errors, please let us know; otherwise, this version will be included in the next release.

PS: A preview version is already available in the internal dev build if you’d like to try it out. Thank you!

``` json
{ "DK", 1, 1, 0, 0, 0, "Nytårsdag", 0, 100, 0, 0 },
{ "DK", 99, -3, 0, 0, 0, "Skærtorsdag", 0, 50, 0, 0 },
{ "DK", 99, -2, 0, 0, 0, "Langfredag", 0, 50, 0, 0 },
{ "DK", 99, 1, 0, 0, 0, "Anden påskedag", 0, 50, 0, 0 },
{ "DK", 99, 26, 0, 0, 0, "Store bededag", 0, 50, 0, 2023 },
{ "DK", 99, 39, 0, 0, 0, "Kristi himmelfartsdag", 0, 50, 0, 0 },
{ "DK", 99, 50, 0, 0, 0, "Anden pinsedag", 0, 50, 0, 0 },
{ "DK", 5, 1, 0, 0, 0, "Arbejdernes kampdag", 0, 100, 0, 0 },
{ "DK", 6, 5, 0, 0, 0, "Grundlovsdag", 0, 50, 0, 0 },
{ "DK", 12, 24, 0, 0, 0, "Juleaftensdag", 0, 50, 0, 0 },
{ "DK", 12, 25, 0, 0, 0, "Juledag", 0, 100, 0, 0 },
{ "DK", 12, 26, 0, 0, 0, "Anden juledag", 0, 100, 0, 0 },
{ "DK", 12, 31, 0, 0, 0, "Nytårsaftensdag", 0, 100, 0, 0 },
```

**Comment by jkoinnovatedk** [comment]
Thank you

I have responded on Github for the three missing days


{ "DK", 1, 1, 0, 0, 0, "Nytårsdag", 0, 100, 0, 0 },

{ "DK", 99, -3, 0, 0, 0, "Skærtorsdag", 0, 50, 0, 0 },

{ "DK", 99, -2, 0, 0, 0, "Langfredag", 0, 50, 0, 0 },

{ "DK", 99, 1, 0, 0, 0, "Anden påskedag", 0, 50, 0, 0 },

{ "DK", 99, 26, 0, 0, 0, "Store bededag", 0, 50, 0, 2023 },

{ "DK", 99, 39, 0, 0, 0, "Kristi himmelfartsdag", 0, 50, 0, 0 },

{ "DK", 99, 50, 0, 0, 0, "Anden pinsedag", 0, 50, 0, 0 },

{ "DK", 5, 1, 0, 0, 0, "Arbejdernes kampdag", 0, 100, 0, 0 },

{ "DK", 6, 5, 0, 0, 0, "Grundlovsdag", 0, 50, 0, 0 },

{ "DK", 12, 24, 0, 0, 0, "Juleaftensdag", 0, 50, 0, 0 },

{ "DK", 12, 25, 0, 0, 0, "Juledag", 0, 100, 0, 0 },

{ "DK", 12, 26, 0, 0, 0, "Anden juledag", 0, 100, 0, 0 },

{ "DK", 12, 31, 0, 0, 0, "Nytårsaftensdag", 0, 100, 0, 0 },

The logic applied in you datetabledefinition does not work with this holidaydefintion  --- ISBLANK (  __HolidayName )

As mentioned in my last reply, some of the dates are not real holidays (non-working days in this case) which should be handled.

So I suggest you delete the following dates from the holidaydefinition (or add an extra 0/1 for “kind of a working holiday” or whatever you decide to call it), since they are normal working days and most people work these days:


§  { "DK", 5, 1, 0, 0, 0, "Arbejdernes kampdag", 0, 100, 0, 0 },

§  { "DK", 6, 5, 0, 0, 0, "Grundlovsdag", 0, 50, 0, 0 },



This in principle also applies to the following but all normal companies are closed these days (non-working days for all except most shops more or less):



§  { "DK", 12, 24, 0, 0, 0, "Juleaftensdag", 0, 50, 0, 0 },

§  { "DK", 12, 31, 0, 0, 0, "Nytårsaftensdag", 0, 100, 0, 0 },



Med venlig hilsen
Innovate Technology

Jørgen Koch
Microsoft Certified Trainer


[innovatelogo-mail]<http://www.innovate.dk/>

Mail      ***@***.******@***.***>
Telefon +45 70 23 00 14
Mobil    +45 20 23 00 14

Se mere på http://www.innovate.dk/

***@***.***

Fra: Alberto Spelta ***@***.***>
Sendt: 24. juli 2024 16:38
Til: sql-bi/Bravo ***@***.***>
Cc: Jørgen Koch ***@***.***>; Mention ***@***.***>
Emne: Re: [sql-bi/Bravo] Danish Holidays (Issue #793)


Thank you for the information you provided. We have updated the names of the currently defined holidays and set 2023 as the last year for Store Bededag. At the moment, there are three holidays that are not included, and we have opened a dedicated issue here<https://github.com/sql-bi/DaxTemplate/issues/53> for these as they likely require a change to the calculation.

For completeness, below is the resulting holiday configuration. If you encounter any errors, please let us know; otherwise, this version will be included in the next release.

PS: A preview version is already available in the internal dev build if you’d like to try it out. Thank you!

{ "DK", 1, 1, 0, 0, 0, "Nytårsdag", 0, 100, 0, 0 },

{ "DK", 99, -3, 0, 0, 0, "Skærtorsdag", 0, 50, 0, 0 },

{ "DK", 99, -2, 0, 0, 0, "Langfredag", 0, 50, 0, 0 },

{ "DK", 99, 1, 0, 0, 0, "Anden påskedag", 0, 50, 0, 0 },

{ "DK", 99, 26, 0, 0, 0, "Store bededag", 0, 50, 0, 2023 },

{ "DK", 99, 39, 0, 0, 0, "Kristi himmelfartsdag", 0, 50, 0, 0 },

{ "DK", 99, 50, 0, 0, 0, "Anden pinsedag", 0, 50, 0, 0 },

{ "DK", 5, 1, 0, 0, 0, "Arbejdernes kampdag", 0, 100, 0, 0 },

{ "DK", 6, 5, 0, 0, 0, "Grundlovsdag", 0, 50, 0, 0 },

{ "DK", 12, 24, 0, 0, 0, "Juleaftensdag", 0, 50, 0, 0 },

{ "DK", 12, 25, 0, 0, 0, "Juledag", 0, 100, 0, 0 },

{ "DK", 12, 26, 0, 0, 0, "Anden juledag", 0, 100, 0, 0 },

{ "DK", 12, 31, 0, 0, 0, "Nytårsaftensdag", 0, 100, 0, 0 },

**Comment by albertospelta** [comment]
Thank you for the info about the three missing holidays and for your help in testing.
I have removed the dates that are not official holidays and below is the updated definition:
```
{ "DK", 1, 1, 0, 0, 0, "Nytårsdag", 0, 100, 0, 0 },
{ "DK", 99, -3, 0, 0, 0, "Skærtorsdag", 0, 50, 0, 0 },
{ "DK", 99, -2, 0, 0, 0, "Langfredag", 0, 50, 0, 0 },
{ "DK", 99, 1, 0, 0, 0, "Anden påskedag", 0, 50, 0, 0 },
{ "DK", 99, 26, 0, 0, 0, "Store bededag", 0, 50, 0, 2023 },
{ "DK", 99, 39, 0, 0, 0, "Kristi himmelfartsdag", 0, 50, 0, 0 },
{ "DK", 99, 50, 0, 0, 0, "Anden pinsedag", 0, 50, 0, 0 },
{ "DK", 12, 25, 0, 0, 0, "Juledag", 0, 100, 0, 0 },
{ "DK", 12, 26, 0, 0, 0, "Anden juledag", 0, 100, 0, 0 },
```

I’m not entirely clear on the issue with the logic in the date table definition and the ISBLANK (__HolidayName) function. Could you please provide more details or an example of the problem? If you could share a sample PBIX file, that would be very helpful for figuring this out.

> The logic applied in you datetabledefinition does not work with this holidaydefintion  --- ISBLANK (  __HolidayName )

**Comment by jkoinnovatedk** [comment]
Hi Alberto

Thank you, it looks good (but still missing the three days, which though always fall on Sundays, and therefore are already non-working days).

What I meant by the ISBLANK (..) comment was that as long as non-working days are determined by whether there is any text in the holiday table, the day (such as Constitution Day, 1 May, Christmas Eve and New Year's Day) should either be removed from the holiday table, as you have done so well), or an extra option should be added in the form of a true-false column,  So you can indicate whether the holiday is actually a working day or a non-working day. Thus, it is not enough that the date appears in the public holiday table to determine whether it is a (genuine) public holiday (a non-working day).

I prefer the chosen solution, where you have removed the dates so everything is fine.

However, it must be said that if you are to use the table to calculate turnover in shops etc., which are open on Christmas Eve and New Year's Day), these two dates must be working days, while in connection with production companies and many others it must be non-working days.

Likewise, many companies have closed on the Friday after Maundy Thursday, but it is again a collective agreement paid day off, so in my opinion it might still be a good idea to have this extra column to indicate whether the so-called public holiday (both official and unofficial) is a working day or not.

In the illustration, for example, both Constitution Day, Christmas Eve (Christmas Eve) and New Year's Eve (New Year) are shown in the calendar, but they are not listed as public holidays (blue marking).
[Et billede, der indeholder tekst, skærmbillede, nummer/tal, Font/skrifttype  Automatisk genereret beskrivelse]    [Et billede, der indeholder tekst, skærmbillede, nummer/tal, Font/skrifttype  Automatisk genereret beskrivelse]




Med venlig hilsen
Innovate Technology

Jørgen Koch
Microsoft Certified Trainer


[innovatelogo-mail]<http://www.innovate.dk/>

Mail      ***@***.******@***.***>
Telefon +45 70 23 00 14
Mobil    +45 20 23 00 14

Se mere på http://www.innovate.dk/

***@***.***

Fra: Alberto Spelta ***@***.***>
Sendt: 25. juli 2024 10:29
Til: sql-bi/Bravo ***@***.***>
Cc: Jørgen Koch ***@***.***>; Mention ***@***.***>
Emne: Re: [sql-bi/Bravo] Danish Holidays (Issue #793)


Thank you for the info about the three missing holidays and for your help in testing.
I have removed the dates that are not official holidays and below is the updated definition:

{ "DK", 1, 1, 0, 0, 0, "Nytårsdag", 0, 100, 0, 0 },

{ "DK", 99, -3, 0, 0, 0, "Skærtorsdag", 0, 50, 0, 0 },

{ "DK", 99, -2, 0, 0, 0, "Langfredag", 0, 50, 0, 0 },

{ "DK", 99, 1, 0, 0, 0, "Anden påskedag", 0, 50, 0, 0 },

{ "DK", 99, 26, 0, 0, 0, "Store bededag", 0, 50, 0, 2023 },

{ "DK", 99, 39, 0, 0, 0, "Kristi himmelfartsdag", 0, 50, 0, 0 },

{ "DK", 99, 50, 0, 0, 0, "Anden pinsedag", 0, 50, 0, 0 },

{ "DK", 12, 25, 0, 0, 0, "Juledag", 0, 100, 0, 0 },

{ "DK", 12, 26, 0, 0, 0, "Anden juledag", 0, 100, 0, 0 },

I’m not entirely clear on the issue with the logic in the date table definition and the ISBLANK (__HolidayName) function. Could you please provide more details or an example of the problem? If you could share a sample PBIX file, that would be very helpful for figuring this out.

The logic applied in you datetabledefinition does not work with this holidaydefintion --- ISBLANK ( __HolidayName )

**Comment by jkoinnovatedk** [comment]
Yeah, but it is actually the other way around.

It could be nice to have some days shown with their responding name, ie. Labour Day, but it still have to be a working day (not a non-working day), but if you put any date with a label in the Holiday definition table, the date will be marked as a non-working date in the date table qua the NOT ISBLANK-logic.

What I was thinking was to expand the Holiday Definition with an extra option marking it as a non-workning day or not:

"Holidays": [
    {
      "IsoCountry": "US",
      "MonthNumber": 1,
      "DayNumber": 1,
      "WeekDayNumber": 0,
      "OffsetWeek": 0,
      "OffsetDays": 0,
      "HolidayName": "New Year's Day",
      "SubstituteHoliday": "NoSubstituteHoliday",
      "ConflictPriority": 100,
      "NonWorkingDay": 1  // this is a non-working day
    },

    {
      "IsoCountry": "DK",
      "MonthNumber": 1,
      "DayNumber": 1,
      "WeekDayNumber": 0,
      "OffsetWeek": 0,
      "OffsetDays": 0,
      "HolidayName": "Grundlovsdag",
      "SubstituteHoliday": "NoSubstituteHoliday",
      "ConflictPriority": 100,
      "NonWorkingDay": 0  // this is a working day
    },


But I suggest we close the topic for now since its only a nice-to-have option, and maybe not that big a problem in the real world (and maybe only for a few Danes).


Med venlig hilsen
Innovate Technology

Jørgen Koch
Microsoft Certified Trainer


[innovatelogo-mail]<http://www.innovate.dk/>

Mail      ***@***.******@***.***>
Telefon +45 70 23 00 14
Mobil    +45 20 23 00 14

Se mere på http://www.innovate.dk/

***@***.***

Fra: Alberto Spelta ***@***.***>
Sendt: 25. juli 2024 17:12
Til: sql-bi/Bravo ***@***.***>
Cc: Jørgen Koch ***@***.***>; Mention ***@***.***>
Emne: Re: [sql-bi/Bravo] Danish Holidays (Issue #793)


@jkoinnovatedk<https://github.com/jkoinnovatedk> To handle specific cases like the ones you mentioned, you can use the Custom Date Templates feature<https://docs.sqlbi.com/bravo/features/manage-dates/customize-date-template>. This feature allows you to clone a standard date template and customize it, for example, by adding unofficial holidays, which can then be marked as non-working days. This should help you manage all the examples you provided.

For more details on how to customize date templates, you can take a look at this video: Custom Date Templates with Bravo for Power BI<https://www.youtube.com/watch?v=pkTmJMd06XI&t=472s>.

**Comment by marcosqlbi** [comment]
@jkoinnovatedk Please, can you describe the use case for holidays that are working days? My concern is that we are going to increase the complexity of the system (which is already non-trivial), making it harder to use for the more common use case, which is just to have the correct working days. For this reason, in general, we tried to avoid adding holidays that are non-working days, as it could generate confusion for certain analyses. At the same time, I understand that the proximity of certain holidays is important for specific products/services, but we didn't receive many requests so far and we try to not increase complexity unless it's necessary.
Thanks!

**Comment by jkoinnovatedk** [comment]
Hi Marco

After thinking more about this matter, I think there will be a very minimal use of this, so to avoid a more complex system, I think we should let it rest.

“we tried to avoid adding holidays that are non-working days” – I hope you mean “… are working days?

Again, I think this will work in 99.4% of the cases – I was just thinking, that some companies would like to add the normally “working holidays” as “non-working days” the get ie. Average Daily Revenue, but since they in the Danish case always are fixed dates, it after thinking about it again actually would be much easier just to add the dates to the HolidayDefinition table in that case.

After next release of Bravo with the Danish holidays incorporated, I will make a LinkedIn tip about how to do this.



Med venlig hilsen
Innovate Technology

Jørgen Koch
Microsoft Certified Trainer


[innovatelogo-mail]<http://www.innovate.dk/>

Mail      ***@***.******@***.***>
Telefon +45 70 23 00 14
Mobil    +45 20 23 00 14

Se mere på http://www.innovate.dk/

***@***.***

Fra: Marco Russo ***@***.***>
Sendt: 26. juli 2024 09:00
Til: sql-bi/Bravo ***@***.***>
Cc: Jørgen Koch ***@***.***>; Mention ***@***.***>
Emne: Re: [sql-bi/Bravo] Danish Holidays (Issue #793)


@jkoinnovatedk<https://github.com/jkoinnovatedk> Please, can you describe the use case for holidays that are working days? My concern is that we are going to increase the complexity of the system (which is already non-trivial), making it harder to use for the more common use case, which is just to have the correct working days. For this reason, in general, we tried to avoid adding holidays that are non-working days, as it could generate confusion for certain analyses. At the same time, I understand that the proximity of certain holidays is important for specific products/services, but we didn't receive many requests so far and we try to not increase complexity unless it's necessary.
Thanks!

**Comment by albertospelta** [comment]
Thanks Jørgen! I'm closing this issue as the translated holiday names have been merged and will be included in the next release.

**Comment by albertospelta** [comment]
@jkoinnovatedk Fixed in version [1.0.9](https://github.com/sql-bi/Bravo/releases/tag/v1.0.9)
